v3/src/components/graph/adornments/univariate-measures/univariate-measure-adornment-helper.ts

```diff
@@ -163,13 +163,12 @@ export class UnivariateMeasureAdornmentHelper {
       ? this.layout.getAxisMultiScale("bottom")
       : this.layout.getAxisMultiScale("left")
     const displayValue = multiScale?.formatValueForScale(value) || valueLabelString(value)
-    const plotValue = Number(displayValue) // Is this really just `value`?
+    const plotValue = Number(displayValue)
     const measureRange: IRange | undefined = attrId && this.model.hasRange
       ? this.model.computeMeasureRange(attrId, this.cellKey, dataConfig)
       : {}
-    const displayRange = measureRange.min || measureRange.min === 0
-      ? multiScale?.formatValueForScale(measureRange.min) || valueLabelString(measureRange.min)
-      : undefined
+    const adjustedValue = value - Number(measureRange?.min)
+    const displayRange = multiScale?.formatValueForScale(adjustedValue) || valueLabelString(adjustedValue)
```

Comment:
1. In line 167, `measureRange` is typed as `IRange | undefined`. It should just be `IRange` because its value is always an object.
2. In line 170, the `?` after `measureRange` is unnecessary since it can't be `undefined`, but `measureRange.min` _can_ be `undefined`, so `Number(measureRange.min)` could be converting `undefined` to a number.
3. It would be useful to add a utility method:
```typescript
  formatValueForScale(isVertical: boolean, value: number | undefined) {
    const multiScale = isVertical
      ? this.layout.getAxisMultiScale("bottom")
      : this.layout.getAxisMultiScale("left")
    return value != null
      ? multiScale
        ? multiScale.formatValueForScale(value)
        : valueLabelString(value)
      : ""
  }
```
4. and then `adornmentSpecs()` would look like this:
```typescript
  adornmentSpecs = (
    attrId: string, dataConfig: IDataConfigurationModel, plotValue: number, isVertical: boolean,
    cellCounts: Record<string, number>, secondaryAxisX=0, secondaryAxisY=0
  ) => {
    const displayValue = this.formatValueForScale(isVertical, plotValue)
    const measureRange: IRange = attrId && this.model.hasRange
      ? this.model.computeMeasureRange(attrId, this.cellKey, dataConfig)
      : {}
    const rangeValue = measureRange.min != null ? plotValue - measureRange.min : undefined
    const displayRange = this.formatValueForScale(isVertical, rangeValue)
    const {x: x1, y: y1} =
      this.calculateLineCoords(plotValue, 1, isVertical, cellCounts, secondaryAxisX, secondaryAxisY)
    const {x: x2, y: y2} =
      this.calculateLineCoords(plotValue, 2, isVertical, cellCounts, secondaryAxisX, secondaryAxisY)
    const lineClass = clsx("measure-line", `${this.measureSlug}-line`)
    const lineId = this.generateIdString("line")
    const coverClass = clsx("measure-cover", `${this.measureSlug}-cover`)
    const coverId = this.generateIdString("cover")

    return {
      coords: {x1, x2, y1, y2},
      coverClass,
      coverId,
      displayRange,
      displayValue,
      lineClass,
      lineId,
      measureRange,
      plotValue
    }
  }
```

Reply:
Thanks. That all makes sense and appears to work great. I should've at least caught my mistaken use of `undefined` for the type in line 167 🤦‍♂️